### PR TITLE
Fix Migration Guide Typo and Add Section on Multiple Deep Link Targets

### LINKED ISSUE #488: Improve Migration Documentation Around Browser Switch Activity
[The migration guide](https://github.com/braintree/braintree_android/blob/master/v4_MIGRATION_GUIDE.md#browser-switch) suggests that we use our own activity rather than V3 `com.braintreepayments.api.BraintreeBrowserSwitchActivity`. But it is unclear what that activity is supposed to do.

`MyPaymentsActivity` referenced in the manifest snippet isn't used anywhere else in the guide.

In my use case I have multiple activities that can create a paypal payment method but I am unsure on how I am supposed to handle this.

### PR DESCRIPTION
**Pull request by sshropshire**
### Summary of changes

 - Fix migration guide typo related to PayPal Browser Switch
 - Add a section to the migration guide for multiple deep-link targets (Fixes #488)

 ### Checklist

 ~- [ ] Added a changelog entry~

### CODE REVIEW

File: v4_MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -53,6 +53,45 @@ In the `AndroidManifest.xml`, migrate the `intent-filter` from your v3 integrati
...
+    </intent-filter>
+</activity>
+``` 
+
+Then when constructing your `BraintreeClient` make sure to pass the appropriate custom url scheme for each deep link target Activity:
+
+```java
+// MyPaymentsActivity1.java
+BraintreeClient braintreeClient =
+        new BraintreeClient(this, "TOKENIZATION_KEY_OR_CLIENT_TOKEN", "custom-url-scheme-1");
```
Comment (sarahkoop) [inline v4_MIGRATION_GUIDE.md:87]:
```suggestion
    new BraintreeClient(this, "TOKENIZATION_KEY_OR_CLIENT_TOKEN", "custom-url-scheme-1");
```

File: v4_MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -488,7 +527,7 @@ public class PayPalActivity extends AppCompatActivity {
       payPalClient.onBrowserSwitchResult(browserSwitchResult, (payPalAccountNonce, error) -> {
         if (payPalAccountNonce != null) {
           // Send nonce to server
-          String nonce = payPalNonce.getString();
+          String payPalAccountNonce = payPalNonce.getString();
```
Comment (sarahkoop) [inline v4_MIGRATION_GUIDE.md:530]:
```suggestion
          String nonce = payPalAccountNonce.getString();
```
I think `payPalNonce` was the typo right?

Comment (sshropshire) [reply v4_MIGRATION_GUIDE.md]: Yikes yeah thanks for that.

**Review by jplukarski — APPROVED** [review]

File: v4_MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -53,6 +53,45 @@ In the `AndroidManifest.xml`, migrate the `intent-filter` from your v3 integrati
 
 Additionally, apps that use both Drop-in and BraintreeClient should specify a custom url scheme, since `DropInActivity` already uses the `${applicationId}.braintree` url intent filter.
 
+If your app has multiple browser switch targets, you can specify multiple intent filters, and use the `BraintreeClient` constructor that allows you to specify a `customUrlScheme`:
```
Comment (cgdibble) [inline v4_MIGRATION_GUIDE.md:56]:
```suggestion
If your app has multiple browser switch targets, you can specify multiple intent filters and use the `BraintreeClient` constructor that allows you to specify a `customUrlScheme`:
```

**Review by cgdibble — APPROVED** [review]
Feel free to disregard the suggestion, too, it's non-blocking!

File: v4_MIGRATION_GUIDE.md (hunk 3 above)
Comment (sshropshire) [reply v4_MIGRATION_GUIDE.md]: @cgdibble good call! Too much comma haha.

**Pull request merged**